resources/js/Components/JobPost/FieldsPart.tsx

```diff
@@ -0,0 +1,150 @@
+import SearchSvg from "/public/assets/images/icon/search-2.svg";
+import CategorySvg from "/public/assets/images/icon/category-2.svg";
+import EmailSvg from "/public/assets/images/icon/email-2.svg";
+
+import SalaryInput from './SalaryInput';
+import InputDate from '../InputDate';
+
+
+const FieldsPart:React.FC<{
+    data:IJobPost,
+    errors:TError,
+    setFieldByKey :(key:TKey,value:string|string[]|[])=>void ,
+    setFieldNewObj:(obj:IJobPost)=>void ,
+}>=({data,errors,setFieldByKey,setFieldNewObj})=>{
+    return(<>
+     <InputText 
+        className="col-md-6"   
+        onChange={e => setFieldByKey(EJobPos.jobtitle, e.target.value)}
+        value={data.jobtitle}
+        errors={errors}
+        name={EJobPos.jobtitle}
+        placeholder='Senior Engineer'
+        srcIcon={CompanySvg}
+        title='Job Title*'
+        />
+        <MySimpleSelect 
+          className="col-md-6"   
+          value={data.jobcategory}
+          name={EJobPos.jobcategory}
+          srcIcon={CategorySvg}
+          title='Job Category*'
+          errors={errors}
+          options={jobCategoryOptions}
+          setField={value => setFieldByKey(EJobPos.jobcategory, value)}
+          />      
+         <InputText 
+        className="col-md-6"   
+        onChange={e => setFieldByKey(EJobPos.vacancies, e.target.value)}
+        value={data.vacancies}
+        errors={errors}
+        name={EJobPos.vacancies}
+        placeholder='7 Person'
+        srcIcon={UserSvg}
+        title='Vacancies*'
+        />
+        <SalaryInput data={data}  setFieldNewObj={setFieldNewObj}/>     
+        <MySimpleSelect 
+        className="col-md-6"  
+        value={data.jobtype}
+        name={EJobPos.jobtype}
+        srcIcon={CompanySvg}
+        title='Job Type*'
+        errors={errors}
+        options={jobTypeOptions}
+        setField={value => setFieldByKey(EJobPos.jobtype, value)}
+        />
+         <MySimpleSelect 
+        className="col-md-6"  
+        name={EJobPos.experiences}
+        value={data.experiences}
+        srcIcon={EmailSvg}
+        title='Experience Level*'
+        errors={errors}
+        options={jobExperiencesOptions}
+        setField={value => setFieldByKey(EJobPos.experiences, value)}
+        />
+         <MyAsyncSelect 
+        className="col-md-6"  
+        name={EJobPos.jobtag}
+        values={data.jobtag}
+        srcIcon={SearchSvg}
+        title='Job Tag*'
+        errors={errors}
+        setField={valueArr => setFieldByKey(EJobPos.jobtag, valueArr?valueArr:[])} 
+        loadOptions={jobTagLoadOptions} 
+        defaultOptions={jobTagOptionsDef}
+        isMulti={true}
+        />
+       <InputDate 
+        className="col-md-6"    
+        name={EJobPos.deadline}
+        title='Deadline*'
+        setField={date=>setFieldByKey(EJobPos.deadline, date)}
+        value={data.deadline}
+       />
+    </>)
+}
+export default FieldsPart;
+
+const jobTypeOptions:  ISelectOptions[] = [
```

Comment:
It's better to move all variables to the top of the component to make it easier to read the code

Reply:
remove  to folder Interfaces